avcodec.cpp

```diff
@@ -406,6 +409,33 @@ static int avcodec_decoder_copy_frame(const avcodec_decoder d, opencv_mat mat, A
             SWS_BILINEAR, // Specify the scaling algorithm; you can choose another according to your needs
             NULL, NULL, NULL);
 
+        // Configure colorspace conversion
+        int colorspace;
+        switch (frame->colorspace)
+        {
+        case AVCOL_PRI_BT2020:
+            colorspace = SWS_CS_BT2020;
+            break;
+        case AVCOL_PRI_BT709:
+            colorspace = SWS_CS_ITU709;
+            break;
+        case AVCOL_PRI_BT470BG:
+        case AVCOL_PRI_SMPTE170M:
+            colorspace = SWS_CS_ITU601;
+            break;
+        default:
+            colorspace = SWS_CS_ITU709;
+            break;
+        }
```

Comment:
This isn't necessary; the `SWS_CS_*` macros correspond to the same integer values as `AVColorSpace` (they both come from ISO/IEC 23091-2:2019 subclause 8.3), and if the value passed to `sws_getCoefficients` is unrecognized or _UNSPECIFIED, you get the BT601 values. I'd say just handle `AVCOL_SPC_UNSPECIFIED` if you wanna make that 709 instead of 601, and otherwise just passthrough the input.

Also, these are the wrong enum values: `colorspace` is `AVColorSpace`/`AVCOL_SPC_*`, not `AVColorPrimaries`/`AVCOL_PRI_*`; they're subtly different and you want the former here.

Reply:
oh wow, yeah, great catch here.